monitoring/etcd.go

```diff
@@ -46,8 +46,14 @@ type ETCDConfig struct {
 	// InsecureSkipVerify controls whether a client verifies the
 	// server's certificate chain and host name.
 	InsecureSkipVerify bool
+	// Client specifies the optional HTTP client to use for checks.
+	// If unspecified, one will be created with default settings
+	Client *http.Client
 }
 
+// defaultHTTPTimeout defines the default HTTP client timeout for HTTP-based checks
+const defaultHTTPTimeout = 10 * time.Second
```

Comment:
More of a curiosity, any insight into why setting 10 seconds as the default or just what seems to make sense?

Reply:
It is more what I thought would make sense - I haven't actually measured the timeouts in those cases where it fails (for example in majority of robotests for master).